modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/ParsedMatrixStats.java

```diff
@@ -0,0 +1,247 @@
+            for (String fieldName : counts.keySet()) {
+                builder.startObject();
+                builder.field(InternalMatrixStats.Fields.NAME, fieldName);
+                builder.field(InternalMatrixStats.Fields.COUNT, getFieldCount(fieldName));
+                builder.field(InternalMatrixStats.Fields.MEAN, getMean(fieldName));
+                builder.field(InternalMatrixStats.Fields.VARIANCE, getVariance(fieldName));
+                builder.field(InternalMatrixStats.Fields.SKEWNESS, getSkewness(fieldName));
+                builder.field(InternalMatrixStats.Fields.KURTOSIS, getKurtosis(fieldName));
+                {
+                    builder.startObject(InternalMatrixStats.Fields.COVARIANCE);
+                    Map<String, Double> covars = covariances.get(fieldName);
+                    if (covars != null) {
+                        for (Map.Entry<String, Double> covar : covars.entrySet()) {
+                            builder.field(covar.getKey(), covar.getValue());
+                        }
+                    }
+                    builder.endObject();
+                }
+                {
+                    builder.startObject(InternalMatrixStats.Fields.CORRELATION);
+                    Map<String, Double> correls = correlations.get(fieldName);
+                    if (correls != null) {
+                        for (Map.Entry<String, Double> correl : correls.entrySet()) {
+                            builder.field(correl.getKey(), correl.getValue());
+                        }
+                    }
+                    builder.endObject();
+                }
+                builder.endObject();
+            }
+            builder.endArray();
+        }
+        return builder;
+    }
+
+    private static <T> T checkedGet(final Map<String, T> values, final String fieldName) {
+        if (fieldName == null) {
+            throw new IllegalArgumentException("field name cannot be null");
+        }
+        if (values.containsKey(fieldName) == false) {
+            throw new IllegalArgumentException("field " + fieldName + " does not exist");
+        }
+        return values.get(fieldName);
+    }
+
+    private static ObjectParser<ParsedMatrixStats, Void> PARSER =
+            new ObjectParser<>(ParsedMatrixStats.class.getSimpleName(), true, ParsedMatrixStats::new);
+    static {
+        declareAggregationFields(PARSER);
+        PARSER.declareObjectArray((matrixStats, results) -> {
+            int size = results.size();
+            matrixStats.counts = new LinkedHashMap<>(size);
+            matrixStats.means = new LinkedHashMap<>(size);
+            matrixStats.variances = new LinkedHashMap<>(size);
+            matrixStats.skewness = new LinkedHashMap<>(size);
+            matrixStats.kurtosis = new LinkedHashMap<>(size);
+            matrixStats.covariances = new LinkedHashMap<>(size);
+            matrixStats.correlations = new LinkedHashMap<>(size);
```

Comment:
No, but at least on `counts` so that we can render the parsed aggregation with the same order of fields stats.